dandi/upload.py

```diff
@@ -413,8 +413,14 @@ def _bids_discover_and_validate(
     from .utils import find_files
     from .validate import validate_bids
 
+    lgr.debug("Discovering root directories of BIDS datasets")
     bids_descriptions = map(
-        Path, find_files(r"(^|[/\x5C])dataset_description\.json$", dandiset_path)
+        Path,
+        find_files(
+            r"(^|[/\x5C])dataset_description\.json$",
+            dandiset_path,
+            dirs_avoid=r"\.(ngff|zarr)(\..*)?$",
```

Comment:
is this bit `(\..*)?` required? I know there is `ome.zarr` — but is there `zarr.ome` or `.ngff.ome`?

Reply:
there was a resolved [comment](https://github.com/dandi/dandi-cli/pull/1071#discussion_r923893998):

ha -- the case at hands has micr/sub-MITU01h3_ses-20211202h20m54s11_sample-20_stain-NN_run-1_chunk-4_SPIM.ngff.source so I would need to add allowing for more extensions for such cases.

Let's add a comment so we know *why* later on

```suggestion
            # for cases like sub-MITU01h3_..._chunk-4_SPIM.ngff.source
            dirs_avoid=r"\.(ngff|zarr)(\..*)?$",
```